### ISSUE TITLE
Allow to specify option values without ":" or "="

### ISSUE THREAD

**Issue by moigagoo** [fenekku/commandeer#10]
Consider the following code:

``` nim
commandline:
  argument path, string
  option target, char, "target", "t"
```

Calling `myapp -t t path` doesn't throw an error but doesn't do the expected thing, because `path` is `t` and `target` is empty. The correct call is `myapp -t:t path` or `myapp -t=t path`.

It should either throw an error or set `target` to `t` in the first case.

**Comment by fenekku** [comment]
Hey, thanks for this interesting observation. It might be possible.  

Right now `commandeer` relies on `parseopt2` to do the initial command line parsing. Great, we don't have to worry about extracting the command line arguments ourself AND developers (and by consequence users) already familiar with `parseopt2`'s behaviour know what to expect and can port their manual usage of `parseopt2` over to `commandeer`! If we are to achieve space separated options, we would have to add a layer of accounting on `parseopt2` to keep track of argument/option order. It is something I have thought a bit about but would need to investigate more.

I am going to keep this open like #5 because they are along the same lines even though I don't have anything in the short term yet.

**Comment by moigagoo** [comment]
>  developers (and by consequence users) already familiar with parseopt2's behaviour know what to expect 

The problem is that people are familiar with CLIs, not the tools to build CLIs :-) And it's quite common to separate options and their values with a space.

Maybe the right thing to do is to add this feature to parseopt2. What do you think?

**Comment by fenekku** [comment]
> add this feature to parseopt2

That would be an interesting thing to do! Getting consensus on a larger user base like parseopt2 / the nim ecosystem may be harder though ;)

**Comment by fenekku** [comment]
Hello! I just merged #16 that addresses this:

```nim
commandline:
  argument path, string
  option target, char, "target", "t"

# called with myapp -t t path
echo "path ", path  # path
echo "target ", target  # t
```

I didn't go the parseopt2 avenue for the stated reasons and also because I suspect parseopt2's current implementation makes it more flexible for another library to work a top of.

In any case I am closing this. Cheers!
